### PR TITLE
Replaced vertx Json with nanojson

### PR DESCRIPTION
**Pull request by adriantodt**
nanojson has its quirks, but I found it was natively Jackson compatible and was easy to convert from vertx ones.

### CODE REVIEW

File: src/main/java/com/mewna/catnip/entity/impl/user/PresenceImpl.java [hunk 1]
```diff
@@ -69,25 +69,25 @@ public void catnip(@Nonnull final Catnip catnip) {
     @Nonnull
     @CheckReturnValue
     public JsonObject asJson() {
-        final JsonObject innerData = new JsonObject()
```
Comment (queer) [inline src/main/java/com/mewna/catnip/entity/impl/user/PresenceImpl.java]: Can you not chain the method calls here?

File: src/main/java/com/mewna/catnip/rest/ResponsePayload.java [hunk 2]
```diff
@@ -27,26 +27,35 @@
...
-import io.vertx.core.buffer.Buffer;
-import io.vertx.core.json.JsonArray;
-import io.vertx.core.json.JsonObject;
+import com.grack.nanojson.JsonArray;
+import com.grack.nanojson.JsonObject;
+import com.grack.nanojson.JsonParser;
+import com.grack.nanojson.JsonParserException;
 
 public class ResponsePayload {
-    private final Buffer buffer;
```
Comment (queer) [inline src/main/java/com/mewna/catnip/rest/ResponsePayload.java]: Did you just replace all uses of Buffer with String?

File: src/main/java/com/mewna/catnip/util/JsonUtil.java [hunk 3]
```diff
@@ -38,6 +40,7 @@
 
 public final class JsonUtil {
     public static final long MAX_SAFE_INTEGER = 9007199254740991L;
+    public static final ObjectMapper MAPPER = new ObjectMapper();
```
Comment (queer) [inline src/main/java/com/mewna/catnip/util/JsonUtil.java]: Is this just for entity de/serialization?

File: src/main/java/com/mewna/catnip/rest/ResponsePayload.java (hunk 2 above)
Comment (adriantodt) [reply src/main/java/com/mewna/catnip/rest/ResponsePayload.java]: only this one

File: src/main/java/com/mewna/catnip/entity/impl/user/PresenceImpl.java (hunk 1 above)
Comment (adriantodt) [reply src/main/java/com/mewna/catnip/entity/impl/user/PresenceImpl.java]: sadly not. gotta use JsonObject.builder() if you want chaining

File: src/main/java/com/mewna/catnip/util/JsonUtil.java (hunk 3 above)
Comment (adriantodt) [reply src/main/java/com/mewna/catnip/util/JsonUtil.java]: yes. just copied JsonObject#mapFrom/mapTo

File: src/main/java/com/mewna/catnip/entity/impl/user/PresenceImpl.java (hunk 1 above)
Comment (queer) [reply src/main/java/com/mewna/catnip/entity/impl/user/PresenceImpl.java]: Imo that would be preferable to this

**Pull request merged**